### DISCUSSION TITLE
Spotify/Artist IDs

### DISCUSSION THREAD

**Discussion by jdg46** [pavelkomarov/exportify#64] [Q&A] [answered]
Hi - big fan of Exportify!

Recently my exports stopped including Spotify ID and Artist ID as the first two fields .. are there new issues with those fields? They're really important! Thanks

**Comment by pavelkomarov** [comment]
Oh, no there's nothing wrong with those fields. I just removed them when I added Record Labels, because they seemed extraneous to me, since they mean basically nothing to a human reader. What do you use them for? If you convince me, I'll add them back; it would take only a few minutes. I was just aiming for simplicity.

**Reply by jdg46** [reply]
I concatenate the trackID with the string "spotify:track:" to make a spotify URI for each track, which enables me to link quickly to a playable track. 

Also I can copy and paste a group URIs right into spotify making easy playlists -- which is critical for my use case of using exportify data to make new 'summary' playlists. For example, i download the data for 30 tastemaker playlists and make a new playlist of the songs with the lowest popularity score that have been added to the most tastemaker playlists. Without the track ID that's just a useless list of names, but the track ID enables to make a playlist of these songs in seconds. So I can't really use Exportify without that field.

Track ID is way more important than Artist ID.

Adding record labels is AWESOME, though. Was my most-desired field (before trackID went away :)

**Comment by pavelkomarov** [comment]
Word. I could also just give you the URI, but the ID is smaller https://developer.spotify.com/documentation/web-api/concepts/spotify-uris-ids. I'll put song ID back in the output. It's just there were song, artist, and album IDs floating around, and that many IDs was starting to look a little ridiculous to me.

**Comment by jdg46** [comment]
Yeah that makes total sense. Really appreciate you putting songID back, definitely don't need the other ones though. Thanks for all you do!

**Comment by pavelkomarov** [comment, marked as answer]
This accomplishes it https://github.com/pavelkomarov/exportify/commit/f0dc2133340eeeed5f2728ee86b56115a9440b72. Should be live momentarily. Cool use case. What a hipster lol. Looks like your GitHub is new, so welcome. Thanks for bothering to go through all those steps just to let me know about this.

**Reply by pavelkomarov** [reply]
I goofed! I also needed to make this change so the leftover ID is for track, not the album! https://github.com/pavelkomarov/exportify/commit/8f37f60e40a0de3917f41e2fe6bb4a140af5d589 It works now.

**Comment by jdg46** [comment]
Yes it does! thank you!
